File: frontend/src/api/export.ts
import api from "./client";

export interface ExportFilter {
  firm_id?: string;
  vendor?: string;
  invoice_number?: string;
  gst_match?: boolean;
  date_from?: string;
  date_to?: string;
}

export async function exportExcel(
  filters: ExportFilter,
): Promise<void> {

  const response = await api.get(
    "/export/excel",
    {
      params: filters,
      responseType: "blob",
    },
  );

  const blob = new Blob([
    response.data,
  ]);

  const url =
    window.URL.createObjectURL(blob);

  const link =
    document.createElement("a");

  link.href = url;

  link.download = "Invoices.xlsx";

  document.body.appendChild(link);

  link.click();

  link.remove();

  window.URL.revokeObjectURL(url);

}

File: backend/app/schemas/export.py
from datetime import date
from uuid import UUID

from fastapi import Query


class ExportFilter:

    def __init__(
        self,
        firm_id: UUID | None = Query(None),
        gst_match: bool | None = Query(None),
        date_from: date | None = Query(None),
        date_to: date | None = Query(None),
        vendor: str | None = Query(None),
        invoice_number: str | None = Query(None),
    ):
        self.firm_id = firm_id
        self.gst_match = gst_match
        self.date_from = date_from
        self.date_to = date_to
        self.vendor = vendor
        self.invoice_number = invoice_number
